### PR TITLE
Dba and dev additions

### PR DESCRIPTION
**Pull request by rcbensley**
Adding clustering and sharding to DBA list.
Adding nosql options to dev list.
Added to several existing categories.

### CODE REVIEW

File: list-dba.md [hunk 1]
```diff
@@ -58,11 +60,12 @@ Articles:
...
 |-----------------------------------------------------------------------|--------------------------------------|-------------------------------------------|-----------------------|
-| [Backup Manager](https://fromdual.com/fromdual-backup-manager-bman)  | YES                                  |                                           | Supports many types of backups. By FromDual. |
+| [Backup Manager](https://fromdual.com/fromdual-backup-manager-bman)   | YES                                  |                                           | Supports many types of backups. By FromDual. |
 | [mariadb-dump](https://mariadb.com/kb/en/mariadb-dump/)               | YES                                  | Distributed with MariaDB                  | Logical backups of data and/or schema. |
 | [Mariabackup](https://mariadb.com/kb/en/mariabackup-overview/)        | YES                                  | Distributed with MariaDB                  | Online physical backups. |
+| [MyDumper](https://github.com/mydumper/mydumper)                      | YES                                  | GPLv3                                     | Parallel logical backups and restores. |
```
Comment (federico-razzoli) [inline list-dba.md:68]: Does it really support MariaDB virtual columns, roles and other MariaDB specific syntax?
Also, with the latest improvements to mariadb-dump, is MyDumper still useful?

File: list-dev.md [hunk 2]
```diff
@@ -110,6 +111,7 @@ Perl
 | PHP                  | [MySQLi](https://www.php.net/manual/en/book.mysqli.php)                                   | MySQL           | Both procedural and OO API. |
 | PHP                  | [PDO MySQL](https://www.php.net/manual/en/ref.pdo-mysql.php)                              | MySQL           | Multi-database abstraction layer. |
 | Python               | [Connector/Python](https://mariadb.com/kb/en/list-of-mariadb-connector-python-releases/)  | YES             |                     |
+| Python               | [PyMySQL](https://github.com/PyMySQL/PyMySQL)                                             | YES             | Pure Python driver, MariaDB >= 10.4 |
```
Comment (federico-razzoli) [inline list-dev.md:114]: As far as I can tell, it doesn't support non-blocking calls and progress reporting. If this is confirmed, I'd set MariaDB Support = 'PARTIAL'.

**Review by federico-razzoli — CHANGES_REQUESTED** [review]
I have some objections and this PR is a bit big. I suggest you remove the new sections from here, and open separate PRs for them.
The others parts LGTM, except a couple of doubts, see the comments.

File: list-dba.md (hunk 1 above)
Comment (rcbensley) [reply list-dba.md:68]: Very valid points. I shall remove this one.

File: list-dba.md [hunk 3]
```diff
@@ -141,9 +141,11 @@ In the list below, SQL-aware means that a proxy understands SQL statements, and
 | Project Name                                                                | MariaDB Support                                 | Platforms             | Free / Commercial  | Notes |
 |-----------------------------------------------------------------------------|-------------------------------------------------|-----------------------|--------------------|-------|
 | [Beekeeper Studio](https://www.beekeeperstudio.io/)                         | [NOT VERIFIED](https://docs.beekeeperstudio.io/user_guide/connecting/first-page/)  | Linux, MacOS, Windows | BOTH               |       |
+| [DataGrip](https://www.jetbrains.com/datagrip/)                             | [YES](https://www.jetbrains.com/datagrip/features/) | Linux, MacOS, Windows | Commercial     |       |
```
Comment (federico-razzoli) [inline list-dba.md:144]: This table should mirror the one in list-dev, so make these changes there too.

File: list-dev.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ This list is intended for Developers. There are lists intended for other audienc
 - [Connectors (Drivers)](https://github.com/Vettabase/awesome-mariadb/blob/main/list-dev.md#connectors-drivers)
 - [IDEs](https://github.com/Vettabase/awesome-mariadb/blob/main/list-dev.md#ides)
 - [ORMs](https://github.com/Vettabase/awesome-mariadb/blob/main/list-dev.md#orms)
+- [NoSQL and Key Value Storage](https://github.com/Vettabase/awesome-mariadb/blob/main/list-dev.md#nosql-and-key-value-storage)
```
Comment (federico-razzoli) [inline list-dev.md:15]: To be moved to a separate PR.

**Review by federico-razzoli — CHANGES_REQUESTED** [review]

**Review by federico-razzoli — APPROVED** [review]

**Pull request merged**